fixture/mock_server.go

```diff
@@ -39,30 +41,30 @@ func StartGrpcTargetTestServer(port int) *grpc.Server {
 // StartHttpTargetTestServer starts a HTTP server on the provided port
 // Optionally, it receives a list of handler functions
 func StartHttpTargetTestServer(pathHandlers []PathResponseHandler) (*http.Server, int) {
-	http.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
+	router := http.NewServeMux()
+
+	router.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
 		// Sleep for half a second to simulate a slow server
 		time.Sleep(500 * time.Millisecond)
 		w.WriteHeader(http.StatusOK)
 	})
 
 	for _, pathHandler := range pathHandlers {
-		http.HandleFunc(pathHandler.Path, pathHandler.PathHandlerFunc)
+		router.HandleFunc(pathHandler.Path, pathHandler.PathHandlerFunc)
 	}
 
-	server := &http.Server{}
-	listener, err := net.Listen("tcp", ":0")
-	if err != nil {
-		panic(err)
+	server := &http.Server{
+		Addr:    ":8080",
```

Comment:
can we not hardcode the port?
otherwise tests will be more flaky when running in multitenant ci agents, and also harder to test locally as most services run on 8080

Reply:
Yes. Was just a test under MacOS to get the tests running without
firewall popup. But it seems test pass independent of the popup.